pvlib/iotools/era5.py

```diff
@@ -0,0 +1,294 @@
+        versions of the variables. See [2]_ for a list of variables names and
+        units.
+
+        Access to the CDS requires user registration, see [6]_. The obtaining
+        API key can either be passed directly to the function or be saved in a
+        local file as described in [3]_.
+
+        It is possible to check your
+        `request status <https://cds.climate.copernicus.eu/cdsapp#!/yourrequests>`_
+        and the `status of all queued requests <https://cds.climate.copernicus.eu/live/queue>`_.
+
+    Parameters
+    ----------
+    latitude: float or list
+        in decimal degrees, between -90 and 90, north is positive (ISO 19115).
+        If latitude is a list, it should have the format [S, N] and
+        latitudes within the range are selected according to the grid.
+    longitude: float or list
+        in decimal degrees, between -180 and 180, east is positive (ISO 19115).
+        If longitude is a list, it should have the format [W, E] and
+        longitudes within the range are selected according to the grid.
+    start: datetime like
+        First day of the requested period
+    end: datetime like
+        Last day of the requested period
+    api_key: str, optional
+        Personal API key for the CDS
+    variables: list, default: ERA5_DEFAULT_VARIABLES
+        List of variables to retrieve (according to CDS naming convention)
+    dataset: str, default 'reanalysis-era5-single-levels'
+        Name of the dataset to retrieve the variables from. Can be either
+        'reanalysis-era5-single-levels' or 'reanalysis-era5-land'.
+    product_type: str, {'reanalysis', 'ensemble_members', 'ensemble_mean',
+                        'ensemble_spread'}, default: 'reanalysis'
+        ERA5 product type
+    grid: list or tuple, default: (0.25, 0.25)
+        User specified grid resolution
+    save_path: str or path-like, optional
+        Filename of where to save data. Should have ".nc" extension.
+    cds_client: CDS API client object, optional
+        CDS API client
+    output_format: {'dataframe', 'dataset'}, optional
+        Type of data object to return. Default is to return a pandas DataFrame
+        if file only contains one location and otherwise return an xarray
+        Dataset.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable ERA5_VARIABLE_MAP.
+
+    Notes
+    -----
+    The returned data includes the following fields by default:
+
+    ========================  ======  =========================================
+    Key, mapped key           Format  Description
+    ========================  ======  =========================================
+    *Mapped field names are returned when the map_variables argument is True*
+    ---------------------------------------------------------------------------
+    2tm, temp_air             float   Air temperature at 2 m above ground (K)
```

Comment:
Should we convert to C?  I forget if other iotools functions do that

Reply:
Also nitpick, some units in this table are in parentheses and some are in brackets

Reply:
I was initially against converting to C as it would get confusing if users imports other temperatures (e.g., dew point temperature) which then still will be in degrees C.

Reply:
I've added a function that converts all variables with units of K to units C. The function is placed in `pvlib.tools` as it will also be used in #1274.